src/Features/VisualBasic/Portable/DesignerAttributes/BasicDesignerAttributeService.vb

```diff
@@ -0,0 +1,54 @@
+﻿' Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+Imports System.Composition
+Imports Microsoft.CodeAnalysis.DesignerAttributes
+Imports Microsoft.CodeAnalysis.Host.Mef
+Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.DesignerAttributes
+    <ExportLanguageService(GetType(IDesignerAttributeService), LanguageNames.VisualBasic), [Shared]>
+    Friend Class BasicDesignerAttributeService
+        Inherits AbstractDesignerAttributeService
+
+        Protected Overrides Function GetAllTopLevelTypeDefined(node As SyntaxNode) As IEnumerable(Of SyntaxNode)
+            Dim compilationUnit = TryCast(node, CompilationUnitSyntax)
```

Comment:
these functions look nearly identical to hte C# versions.  can we not share much more code in the base type?